<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Network Centrality Metrics for Elastic-Net Regularized Models • glmSparseNet</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Network Centrality Metrics for Elastic-Net Regularized Models">
<meta property="og:description" content='glmSparseNet is an R-package that generalizes sparse
    regression models when the features (e.g. genes) have a graph
    structure (e.g. protein-protein interactions), by including
    network-based regularizers.  glmSparseNet uses the glmnet R-package,
    by including centrality measures of the network as penalty weights in
    the regularization. The current version implements regularization
    based on node degree, i.e. the strength and/or number of its
    associated edges, either by promoting hubs in the solution or orphan
    genes in the solution. All the glmnet distribution families are
    supported, namely "gaussian", "poisson", "binomial", "multinomial",
    "cox", and "mgaussian".'>
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">glmSparseNet</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.21.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/example_brca_logistic.html">Example for Classification Data -- Breast Invasive Carcinoma</a>
    </li>
    <li>
      <a href="articles/example_brca_protein-protein-interactions_survival.html">Breast survival dataset using network from STRING DB</a>
    </li>
    <li>
      <a href="articles/example_brca_survival.html">Example for Survival Data -- Breast Invasive Carcinoma</a>
    </li>
    <li>
      <a href="articles/example_prad_survival.html">Example for Survival Data -- Prostate Adenocarcinoma</a>
    </li>
    <li>
      <a href="articles/example_skcm_survival.html">Example for Survival Data -- Skin Melanoma</a>
    </li>
    <li>
      <a href="articles/separate2GroupsCox.html">Separate 2 groups in Cox regression</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="glmsparsenet">glmSparseNet<a class="anchor" aria-label="anchor" href="#glmsparsenet"></a>
</h1></div>
<ul>
<li><a href="#overview">Overview</a></li>
<li><a href="#citation">Citation</a></li>
<li><a href="#instalation">Instalation</a></li>
<li>
<a href="#details">Details</a>
<ul>
<li><a href="#function-definition">Function definition</a></li>
</ul>
</li>
<li><a href="#example-for-survival-analysis-using-rna-seq-data">Example for survival analysis using RNA-seq data</a></li>
<li>
<a href="#visualization-and-analytical-tools">Visualization and Analytical tools</a>
<ul>
<li><a href="#survival-curves-with-separate2groupscox">Survival curves with <code>separate2groupsCox</code></a></li>
</ul>
</li>
</ul>
<!-- README.md and README.html are generated from README.Rmd. Please edit that file --><blockquote>
<p>Elastic-Net models with additional regularization based on network centrality metrics</p>
</blockquote>
<p><a href="https://github.com/sysbiomed/glmSparseNet/actions/workflows/check.yml" class="external-link"><img src="https://github.com/sysbiomed/glmSparseNet/actions/workflows/check.yml/badge.svg" alt="R-CMD-check"></a> <a href="https://codecov.io/github/sysbiomed/glmSparseNet" class="external-link"><img src="https://codecov.io/github/sysbiomed/glmSparseNet/graph/badge.svg?token=spsEKgAbyi" alt="codecov"></a></p>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p><code>glmSparseNet</code> is a R package that generalizes sparse regression models when the features <em>(e.g. genes)</em> have a graph structure <em>(e.g. protein-protein interactions)</em>, by including network-based regularizers. <code>glmSparseNet</code> uses the <code>glmnet</code> R-package, by including centrality measures of the network as penalty weights in the regularization. The current version implements regularization based on node degree, i.e. the strength and/or number of its associated edges, either by promoting hubs in the solution or orphan genes in the solution. All the <code>glmnet</code> distribution families are supported, namely <em>“gaussian”</em>, <em>“poisson”</em>, <em>“binomial”</em>, <em>“multinomial”</em>, <em>“cox”</em>, and <em>“mgaussian”</em>.</p>
<p>It adds two new main functions called <code>glmSparseNet</code> and <code>cv.glmSparseNet</code> that extend both model inference and model selection via cross-validation with network-based regularization. These functions are very flexible and allow to transform the penalty weights after the centrality metric is calculated, thus allowing to change how it affects the regularization. To facilitate users, we made available a function that will penalize low connected nodes in the network - <code>glmHub</code> or <code>glmDegree</code> - and another that will penalize hubs - <code>glmOrphan</code>.</p>
<p><span style="display:block;text-align:center"><img src="inst/images/overview.png" alt="Overview of the R-Package pipeline"></span></p>
<p>Below, we provide one example for survival analysis using transcriptomic data from the TCGA Adrenocortical Carcinoma project. More information and Rmd files are available in the vignettes folder where more extensive and complete examples are provided for logistic regresson and Cox’s regression for different types of cancer data.</p>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<p>Veríssimo, A., Carrasquinha E., Lopes, M.B., Oliveira, A.L., Sagot, M.-F. &amp; Vinga, S. (2018), Sparse network-based regularization for the analysis of patientomics high-dimensional survival data. bioRxiv 403402; doi: <a href="https://doi.org/10.1101/403402" class="external-link uri">https://doi.org/10.1101/403402</a></p>
<p>Veríssimo, A., Oliveira, A.L., Sagot, M.-F., &amp; Vinga, S. (2016). DegreeCox – a network-based regularization method for survival analysis. BMC Bioinformatics. 17(16): 449. <a href="https://doi.org/10.1186/s12859-016-1310-4" class="external-link uri">https://doi.org/10.1186/s12859-016-1310-4</a></p>
<p>This package was developed by André Veríssimo, Eunice Carrasquinha, Marta B. Lopes and Susana Vinga under the project SOUND, funded from the European Union Horizon 2020 research and innovation program under grant agreement No. 633974.</p>
</div>
<div class="section level2">
<h2 id="instalation">Instalation<a class="anchor" aria-label="anchor" href="#instalation"></a>
</h2>
<p>Bioconductor is necessary for the installation of this package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"glmSparseNet"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a>
</h2>
<p>This package extends the <code>glmnet</code> r-package with network-based regularization based on features relations. This network can be calculated from the data itself or using external networks to enrich the model.</p>
<p>There are 2 methods available to use data-dependant methods to generate the network:</p>
<ol style="list-style-type: decimal">
<li>Correlation matrix with cutoff;</li>
<li>Covariance matrix with cutoff;</li>
</ol>
<p>Alternatively, the network can be passed as an adjancency matrix or an already calculate metric for each node.</p>
<div class="section level3">
<h3 id="function-definition">Function definition<a class="anchor" aria-label="anchor" href="#function-definition"></a>
</h3>
<p>The main functions from this packages are the <code>glmSparseNet</code> and <code>cv.glmSparseNet</code> and the arguments for the functions are defined as:</p>
<ul>
<li>
<code>xdata</code>: A MultiAssayExperiment object or an input matrix of dimension <code>Observations x Features</code>
</li>
<li>
<code>ydata</code>: Response object that can take different forms depending on the model family that is used</li>
<li>
<code>family</code>: Model type that can take: <em>“gaussian”</em>, <em>“poisson”</em>, <em>“binomial”</em>, <em>“multinomial”</em>, <em>“cox”</em>, and <em>“mgaussian”</em>
</li>
<li>
<code>network</code>: Network to use in penalization, it can take as input: “correlation”, “covariance”, a matrix object with p.vars x p.vars representing the network, a weighted vector of penalties</li>
<li>
<code>experiment.name</code>: Optional parameter used with a “MultiAssayExperiment” object as input</li>
<li>
<code>network.options</code>: Optional parameter defining the options to process the network, such as:
<ul>
<li>
<code>cutoff</code>: A real number to use to remove edges from the network</li>
<li>
<code>minDegree</code>: Minimum value that the weight should have, this is useful as when the weight is 0, there is no regularization on that feature, which may lead to convergence problems</li>
<li>
<code>transFun</code>: Transformation function to the vector of penalty weights after these are calculated from the network</li>
</ul>
</li>
</ul>
<p><em>note:</em> These functions can take any additional arguments that <code>glmnet</code> or <code>cv.glmnet</code> accept (e.g. number of folds in cross validation)</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/cv.glmSparseNet.html">cv.glmSparseNet</a></span><span class="op">(</span></span>
<span>  <span class="va">xdata</span>,</span>
<span>  <span class="va">ydata</span>,</span>
<span>  family <span class="op">=</span> <span class="st">"cox"</span>,</span>
<span>  network <span class="op">=</span> <span class="st">"correlation"</span>,</span>
<span>  options <span class="op">=</span> <span class="fu"><a href="reference/networkOptions.html">networkOptions</a></span><span class="op">(</span></span>
<span>    cutoff <span class="op">=</span> <span class="fl">.6</span>,</span>
<span>    minDegree <span class="op">=</span> <span class="fl">0.2</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="example-for-survival-analysis-using-rna-seq-data">Example for survival analysis using RNA-seq data<a class="anchor" aria-label="anchor" href="#example-for-survival-analysis-using-rna-seq-data"></a>
</h2>
<p>This example uses an adrenal cancer dataset using the correlation to calculate the network and cross-validation to find the optimal model. The network itself if filtered using a cutoff value of 0.6, i.e. all edges that have a correlation between the two features <em>(genes)</em> below the cutoff value are discarded.</p>
<p>The data was retrieved from TCGA database and the Adrenocortical Carcinoma project with 92 patients and a reduced RNASeq data. See Bioconductor package <code>MultiAssayExperiment</code> for more information on the <code>miniACC</code> dataset.</p>
<p>To run the following examples, the next libraries are also needed:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">futile.logger</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/hadley/reshape" class="external-link">reshape2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://waldronlab.io/MultiAssayExperiment/" class="external-link">MultiAssayExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/therneau/survival" class="external-link">survival</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://glmnet.stanford.edu" class="external-link">glmnet</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://www.github.com/sysbiomed/glmSparseNet" class="external-link">glmSparseNet</a></span><span class="op">)</span></span></code></pre></div>
<p>There is some pre-processing needed to remove patients with invalid follow-up date or death date:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"miniACC"</span>, package <span class="op">=</span> <span class="st">"MultiAssayExperiment"</span><span class="op">)</span></span>
<span><span class="va">xdata</span> <span class="op">&lt;-</span> <span class="va">miniACC</span></span>
<span></span>
<span><span class="co"># build valid data with days of last follow up or to event</span></span>
<span><span class="va">eventIx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">xdata</span><span class="op">$</span><span class="va">days_to_death</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">censIx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">xdata</span><span class="op">$</span><span class="va">days_to_last_followup</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">survEventTime</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/array.html" class="external-link">array</a></span><span class="op">(</span><span class="cn">NA</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">xdata</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">survEventTime</span><span class="op">[</span><span class="va">eventIx</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">xdata</span><span class="op">$</span><span class="va">days_to_death</span><span class="op">[</span><span class="va">eventIx</span><span class="op">]</span></span>
<span><span class="va">survEventTime</span><span class="op">[</span><span class="va">censIx</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">xdata</span><span class="op">$</span><span class="va">days_to_last_followup</span><span class="op">[</span><span class="va">censIx</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Keep only valid individuals</span></span>
<span><span class="co">#</span></span>
<span><span class="co"># they are valid if they have:</span></span>
<span><span class="co">#  - either a follow_up time or event time</span></span>
<span><span class="co">#  - a valid vital_status (i.e. not missing)</span></span>
<span><span class="co">#  - folloup_time or event_time &gt; 0</span></span>
<span><span class="va">validIx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/vector.html" class="external-link">as.vector</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">survEventTime</span><span class="op">)</span> <span class="op">&amp;</span> <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">xdata</span><span class="op">$</span><span class="va">vital_status</span><span class="op">)</span> <span class="op">&amp;</span> <span class="va">survEventTime</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span></span>
<span><span class="va">ydata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  time <span class="op">=</span> <span class="va">survEventTime</span><span class="op">[</span><span class="va">validIx</span><span class="op">]</span>,</span>
<span>  status <span class="op">=</span> <span class="va">xdata</span><span class="op">$</span><span class="va">vital_status</span><span class="op">[</span><span class="va">validIx</span><span class="op">]</span>,</span>
<span>  row.names <span class="op">=</span> <span class="va">xdata</span><span class="op">$</span><span class="va">patientID</span><span class="op">[</span><span class="va">validIx</span><span class="op">]</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>The function <code>cv.glmSparseNet</code> fits the survival data using 10-fold cross validation and using a cutoff value of 0.6 to reduce the size of the network.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># build response object for glmnet</span></span>
<span><span class="va">fit3</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/cv.glmSparseNet.html">cv.glmSparseNet</a></span><span class="op">(</span></span>
<span>  <span class="va">xdata</span>, </span>
<span>  <span class="va">ydata</span>,</span>
<span>  family <span class="op">=</span> <span class="st">"cox"</span>,</span>
<span>  network <span class="op">=</span> <span class="st">"correlation"</span>,</span>
<span>  experiment <span class="op">=</span> <span class="st">"RNASeq2GeneNorm"</span>,</span>
<span>  alpha <span class="op">=</span> <span class="fl">.7</span>,</span>
<span>  nlambda <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  options <span class="op">=</span> <span class="fu"><a href="reference/networkOptions.html">networkOptions</a></span><span class="op">(</span></span>
<span>    cutoff <span class="op">=</span> <span class="fl">.6</span>,</span>
<span>    minDegree <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>    transFun <span class="op">=</span> <span class="va">hubHeuristic</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Warning: 'experiments' dropped; see 'drops()'</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="do">## harmonizing input:</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="do">##   removing 306 sampleMap rows not in names(experiments)</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="do">##   removing 13 colData rownames not in sampleMap 'primary'</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">fit3</span><span class="op">)</span></span></code></pre></div>
<p><img src="README_files/figure-gfm/fit.surv-1.png"><!-- --></p>
<p><em>Cross validation plot, showing all 1000 lambdas tested and the error for each, vertical lines show best model and another with fewer variables selected within one standard error of the best.</em></p>
</div>
<div class="section level2">
<h2 id="visualization-and-analytical-tools">Visualization and Analytical tools<a class="anchor" aria-label="anchor" href="#visualization-and-analytical-tools"></a>
</h2>
<div class="section level3">
<h3 id="survival-curves-with-separate2groupscox">Survival curves with <code>separate2groupsCox</code>
<a class="anchor" aria-label="anchor" href="#survival-curves-with-separate2groupscox"></a>
</h3>
<p>This function generates Kaplan-Meier survival model based on the estimated coefficients of the Cox model. It creates two groups based on the relative risk and displays both survival curves <em>(high vs. low-risk patients, as defined by the median)</em> and the corresponding results of log-rank tests.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Data to use in draw.kaplan function</span></span>
<span><span class="co">#  * it takes the input data, response and coefficients</span></span>
<span><span class="co">#  * calculates the relative risk</span></span>
<span><span class="co">#  * separates individuals based on relative risk into High/Low risk groups</span></span>
<span><span class="va">xdataReduced</span> <span class="op">&lt;-</span> <span class="fu">as</span><span class="op">(</span><span class="va">xdata</span><span class="op">[</span>, , <span class="st">"RNASeq2GeneNorm"</span><span class="op">]</span>, <span class="st">"MatchedAssayExperiment"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Warning: 'experiments' dropped; see 'drops()'</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="do">## harmonizing input:</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="do">##   removing 306 sampleMap rows not in names(experiments)</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="do">##   removing 13 colData rownames not in sampleMap 'primary'</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ydataKM</span> <span class="op">&lt;-</span> <span class="va">ydata</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">xdataReduced</span><span class="op">)</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="va">bestModelCoef</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/coef.html" class="external-link">coef</a></span><span class="op">(</span><span class="va">fit3</span>, s <span class="op">=</span> <span class="st">"lambda.min"</span><span class="op">)</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span></span></code></pre></div>
<p>Kaplan-Meier plot</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/separate2GroupsCox.html">separate2GroupsCox</a></span><span class="op">(</span></span>
<span>  <span class="va">bestModelCoef</span>, <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">xdata</span><span class="op">[[</span><span class="st">"RNASeq2GeneNorm"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>, <span class="va">ydataKM</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Coordinate system already present. Adding new coordinate system, which will</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="do">## replace the existing one.</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="do">## $pvalue</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] 2.728306e-07</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="do">## $plot</span></span></code></pre></div>
<p><img src="README_files/figure-gfm/call.kaplan-1.png"><!-- --></p>
<div class="sourceCode" id="cb13"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="do">## $km</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Call: survfit(formula = survival::Surv(time, status) ~ group, data = prognosticIndexDf)</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="do">##                n events median 0.95LCL 0.95UCL</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="do">## Low risk - 1  40      5     NA      NA      NA</span></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a><span class="do">## High risk - 1 39     23   1105     579      NA</span></span></code></pre></div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://www.bioconductor.org/packages/glmSparseNet" class="external-link">View on Bioconductor</a></li>
<li><a href="https://www.github.com/sysbiomed/glmSparseNet/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing glmSparseNet</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>André Veríssimo <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-2212-339X" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Susana Vinga <br><small class="roles"> Author </small>  </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by André Veríssimo, Susana Vinga.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
