<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Breast survival dataset using network from STRING DB • glmSparseNet</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Breast survival dataset using network from STRING DB">
<meta property="og:description" content="glmSparseNet">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">glmSparseNet</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.21.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/example_brca_logistic.html">Example for Classification Data -- Breast Invasive Carcinoma</a>
    </li>
    <li>
      <a href="../articles/example_brca_protein-protein-interactions_survival.html">Breast survival dataset using network from STRING DB</a>
    </li>
    <li>
      <a href="../articles/example_brca_survival.html">Example for Survival Data -- Breast Invasive Carcinoma</a>
    </li>
    <li>
      <a href="../articles/example_prad_survival.html">Example for Survival Data -- Prostate Adenocarcinoma</a>
    </li>
    <li>
      <a href="../articles/example_skcm_survival.html">Example for Survival Data -- Skin Melanoma</a>
    </li>
    <li>
      <a href="../articles/separate2GroupsCox.html">Separate 2 groups in Cox regression</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Breast survival dataset using network from
STRING DB</h1>
                        <h4 data-toc-skip class="author">André
Veríssimo</h4>
            
            <h4 data-toc-skip class="date">2024-02-14</h4>
      
      
      <div class="hidden name"><code>example_brca_protein-protein-interactions_survival.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="instalation">Instalation<a class="anchor" aria-label="anchor" href="#instalation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"glmSparseNet"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="required-packages">Required Packages<a class="anchor" aria-label="anchor" href="#required-packages"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://Matrix.R-forge.R-project.org" class="external-link">Matrix</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://forcats.tidyverse.org/" class="external-link">forcats</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">parallel</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/hadley/reshape" class="external-link">reshape2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/therneau/survival" class="external-link">survival</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">VennDiagram</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">futile.logger</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">curatedTCGAData</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://waldronlab.io/MultiAssayExperiment/" class="external-link">MultiAssayExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">TCGAutils</span><span class="op">)</span></span>
<span><span class="co">#</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://www.github.com/sysbiomed/glmSparseNet" class="external-link">glmSparseNet</a></span><span class="op">)</span></span>
<span><span class="co">#</span></span>
<span><span class="co">#</span></span>
<span><span class="co"># Some general options for futile.logger the debugging package</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/futile.logger/man/flog.layout.html" class="external-link">flog.layout</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/futile.logger/man/flog.layout.html" class="external-link">layout.format</a></span><span class="op">(</span><span class="st">"[~l] ~m"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span></span>
<span>    <span class="st">"glmSparseNet.show_message"</span> <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    <span class="st">"glmSparseNet.base_dir"</span> <span class="op">=</span> <span class="fu">withr</span><span class="fu">::</span><span class="fu"><a href="https://withr.r-lib.org/reference/with_tempfile.html" class="external-link">local_tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co"># Setting ggplot2 default theme as minimal</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme_get.html" class="external-link">theme_set</a></span><span class="op">(</span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>This vignette uses the STRING database (<a href="https://string-db.org/" class="external-link uri">https://string-db.org/</a>) of
protein-protein interactions as the network-based penalizer in
generalized linear models using Breast invasive carcinoma sample
dataset.</p>
<p>The degree vector is calculated manually to account for genes that
are not present in the STRING database, as these will not have any
interactions, i.e. edges.</p>
<div class="section level3">
<h3 id="download-data-from-string">Download Data from STRING<a class="anchor" aria-label="anchor" href="#download-data-from-string"></a>
</h3>
<p>Retrieve all interactions from <a href="https://string-db.org/" class="external-link">STRING databse</a>. We have included a
helper function that retrieves the Homo sapiens known interactions.</p>
<p>For this vignette, we use a cached version of all interaction with
<code>score_threshold = 700</code></p>
<p><em>Note</em>: Text-based interactions are excluded from the
network.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Not evaluated in vignette as it takes too long to download and process</span></span>
<span><span class="va">allInteractions700</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/stringDBhomoSapiens.html">stringDBhomoSapiens</a></span><span class="op">(</span>scoreThreshold <span class="op">=</span> <span class="fl">700</span><span class="op">)</span></span>
<span><span class="va">stringNetwork</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/buildStringNetwork.html">buildStringNetwork</a></span><span class="op">(</span><span class="va">allInteractions700</span>, <span class="st">"external"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="build-network-matrix">Build network matrix<a class="anchor" aria-label="anchor" href="#build-network-matrix"></a>
</h2>
<p>Build a sparse matrix object that contains the network.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">stringNetworkUndirected</span> <span class="op">&lt;-</span> <span class="va">stringNetwork</span> <span class="op">+</span> <span class="fu">Matrix</span><span class="fu">::</span><span class="fu">t</span><span class="op">(</span><span class="va">stringNetwork</span><span class="op">)</span></span>
<span><span class="va">stringNetworkUndirected</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="va">stringNetworkUndirected</span> <span class="op">!=</span> <span class="fl">0</span><span class="op">)</span> <span class="op">*</span> <span class="fl">1</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="network-statistics">Network Statistics<a class="anchor" aria-label="anchor" href="#network-statistics"></a>
</h2>
<div class="section level3">
<h3 id="graph-information">Graph information<a class="anchor" aria-label="anchor" href="#graph-information"></a>
</h3>
<pre><code><span><span class="co">## [INFO] Directed graph (score_threshold = 700)</span></span>
<span><span class="co">## [INFO]   *       total edges: 225330</span></span>
<span><span class="co">## [INFO]   *    unique protein: 11033</span></span>
<span><span class="co">## [INFO]   * edges per protein: 20.423276</span></span>
<span><span class="co">## [INFO]</span></span>
<span><span class="co">## [INFO] Undirected graph (score_threshold = 700)</span></span>
<span><span class="co">## [INFO]   *       total edges: 225209</span></span>
<span><span class="co">## [INFO]   *    unique protein: 11033</span></span>
<span><span class="co">## [INFO]   * edges per protein: 20.412309</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="summary-of-degree-indegree-outdegree">Summary of degree <em>(indegree + outdegree)</em><a class="anchor" aria-label="anchor" href="#summary-of-degree-indegree-outdegree"></a>
</h3>
<pre><code><span><span class="co">## [INFO] Summary of degree:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></span>
<span><span class="co">##    1.00    3.00   13.00   40.83   43.00 4175.00</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="histogram-of-degree-up-until-99-999-quantile">Histogram of degree <em>(up until 99.999% quantile)</em><a class="anchor" aria-label="anchor" href="#histogram-of-degree-up-until-99-999-quantile"></a>
</h3>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/qplot.html" class="external-link">qplot</a></span><span class="op">(</span></span>
<span>    <span class="va">degreeNetworkVector</span><span class="op">[</span></span>
<span>        <span class="va">degreeNetworkVector</span> <span class="op">&lt;=</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">degreeNetworkVector</span>, probs <span class="op">=</span> <span class="fl">.99999</span><span class="op">)</span></span>
<span>    <span class="op">]</span>,</span>
<span>    geom <span class="op">=</span> <span class="st">"histogram"</span>, fill <span class="op">=</span> <span class="fu"><a href="../reference/myColors.html">my.colors</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span>, bins <span class="op">=</span> <span class="fl">100</span></span>
<span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"Degree (up until 99.999% quantile)"</span><span class="op">)</span></span></code></pre></div>
<p><img src="example_brca_protein-protein-interactions_survival_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="glmsparsenet">
<code>glmSparseNet</code><a class="anchor" aria-label="anchor" href="#glmsparsenet"></a>
</h2>
<ul>
<li>Dataset from curatedTCGAdata</li>
</ul>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">brca</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/curatedTCGAData/man/curatedTCGAData.html" class="external-link">curatedTCGAData</a></span><span class="op">(</span></span>
<span>    diseaseCode <span class="op">=</span> <span class="st">"BRCA"</span>, assays <span class="op">=</span> <span class="st">"RNASeq2GeneNorm"</span>,</span>
<span>    version <span class="op">=</span> <span class="st">"1.1.38"</span>, dry.run <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Build the survival data from the clinical columns.</p>
<ul>
<li>Selects only primary solid tumour samples</li>
<li>Merge survival times for patients, which have different columns in
case they are alive or dead.</li>
<li>Build two matrix objects that fit the data <code>xdata</code> and
<code>ydata</code>
</li>
</ul>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># keep only solid tumour (code: 01)</span></span>
<span><span class="va">brcaPrimarySolidTumor</span> <span class="op">&lt;-</span> <span class="fu">TCGAutils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/TCGAutils/man/curatedTCGAData-helpers.html" class="external-link">TCGAsplitAssays</a></span><span class="op">(</span><span class="va">brca</span>, <span class="st">"01"</span><span class="op">)</span></span>
<span><span class="va">xdataRaw</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">brcaPrimarySolidTumor</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get survival information</span></span>
<span><span class="va">ydataRaw</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">brcaPrimarySolidTumor</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="co"># Convert days to integer</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span>        Days.to.date.of.Death <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html" class="external-link">as.integer</a></span><span class="op">(</span><span class="va">Days.to.date.of.Death</span><span class="op">)</span>,</span>
<span>        Days.to.Last.Contact  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html" class="external-link">as.integer</a></span><span class="op">(</span><span class="va">Days.to.Date.of.Last.Contact</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="co"># Find max time between all days (ignoring missings)</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rowwise.html" class="external-link">rowwise</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span>        time <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">days_to_last_followup</span>, <span class="va">Days.to.date.of.Death</span>,</span>
<span>            <span class="va">Days.to.Last.Contact</span>, <span class="va">days_to_death</span>,</span>
<span>            na.rm <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>        <span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="co"># Keep only survival variables and codes</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">patientID</span>, status <span class="op">=</span> <span class="va">vital_status</span>, <span class="va">time</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="co"># Discard individuals with survival time less or equal to 0</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">time</span><span class="op">)</span> <span class="op">&amp;</span> <span class="va">time</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Set index as the patientID</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">ydataRaw</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">ydataRaw</span><span class="op">$</span><span class="va">patientID</span></span>
<span></span>
<span><span class="co"># keep only features that are in degreeNetworkVector and have</span></span>
<span><span class="co">#  standard deviation &gt; 0</span></span>
<span><span class="va">validFeatures</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">xdataRaw</span><span class="op">)</span><span class="op">[</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">xdataRaw</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">degreeNetworkVector</span><span class="op">[</span><span class="va">degreeNetworkVector</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="op">]</span></span>
<span><span class="va">xdataRaw</span> <span class="op">&lt;-</span> <span class="va">xdataRaw</span><span class="op">[</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/TCGAutils/man/TCGAbarcode.html" class="external-link">TCGAbarcode</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">xdataRaw</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">ydataRaw</span><span class="op">)</span>, <span class="va">validFeatures</span></span>
<span><span class="op">]</span></span>
<span><span class="va">xdataRaw</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/scale.html" class="external-link">scale</a></span><span class="op">(</span><span class="va">xdataRaw</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Order ydata the same as assay</span></span>
<span><span class="va">ydataRaw</span> <span class="op">&lt;-</span> <span class="va">ydataRaw</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/TCGAutils/man/TCGAbarcode.html" class="external-link">TCGAbarcode</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">xdataRaw</span><span class="op">)</span><span class="op">)</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co"># Using only a subset of genes previously selected to keep this short example.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="va">params</span><span class="op">$</span><span class="va">seed</span><span class="op">)</span></span>
<span><span class="va">smallSubset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="st">"AAK1"</span>, <span class="st">"ADRB1"</span>, <span class="st">"AK7"</span>, <span class="st">"ALK"</span>, <span class="st">"APOBEC3F"</span>, <span class="st">"ARID1B"</span>, <span class="st">"BAMBI"</span>,</span>
<span>    <span class="st">"BRAF"</span>, <span class="st">"BTG1"</span>, <span class="st">"CACNG8"</span>, <span class="st">"CASP12"</span>, <span class="st">"CD5"</span>, <span class="st">"CDA"</span>, <span class="st">"CEP72"</span>,</span>
<span>    <span class="st">"CPD"</span>, <span class="st">"CSF2RB"</span>, <span class="st">"CSN3"</span>, <span class="st">"DCT"</span>, <span class="st">"DLG3"</span>, <span class="st">"DLL3"</span>, <span class="st">"DPP4"</span>,</span>
<span>    <span class="st">"DSG1"</span>, <span class="st">"EDA2R"</span>, <span class="st">"ERP27"</span>, <span class="st">"EXD1"</span>, <span class="st">"GABBR2"</span>, <span class="st">"GADD45A"</span>,</span>
<span>    <span class="st">"GBP1"</span>, <span class="st">"HTR1F"</span>, <span class="st">"IFNK"</span>, <span class="st">"IRF2"</span>, <span class="st">"IYD"</span>, <span class="st">"KCNJ11"</span>, <span class="st">"KRTAP5-6"</span>,</span>
<span>    <span class="st">"MAFA"</span>, <span class="st">"MAGEB4"</span>, <span class="st">"MAP2K6"</span>, <span class="st">"MCTS1"</span>, <span class="st">"MMP15"</span>, <span class="st">"MMP9"</span>,</span>
<span>    <span class="st">"NFKBIA"</span>, <span class="st">"NLRC4"</span>, <span class="st">"NT5C1A"</span>, <span class="st">"OPN4"</span>, <span class="st">"OR13C5"</span>, <span class="st">"OR13C8"</span>,</span>
<span>    <span class="st">"OR2T6"</span>, <span class="st">"OR4K2"</span>, <span class="st">"OR52E6"</span>, <span class="st">"OR5D14"</span>, <span class="st">"OR5H1"</span>, <span class="st">"OR6C4"</span>,</span>
<span>    <span class="st">"OR7A17"</span>, <span class="st">"OR8J3"</span>, <span class="st">"OSBPL1A"</span>, <span class="st">"PAK6"</span>, <span class="st">"PDE11A"</span>, <span class="st">"PELO"</span>,</span>
<span>    <span class="st">"PGK1"</span>, <span class="st">"PIK3CB"</span>, <span class="st">"PMAIP1"</span>, <span class="st">"POLR2B"</span>, <span class="st">"POP1"</span>, <span class="st">"PPFIA3"</span>,</span>
<span>    <span class="st">"PSME1"</span>, <span class="st">"PSME2"</span>, <span class="st">"PTEN"</span>, <span class="st">"PTGES3"</span>, <span class="st">"QARS"</span>, <span class="st">"RABGAP1"</span>,</span>
<span>    <span class="st">"RBM3"</span>, <span class="st">"RFC3"</span>, <span class="st">"RGPD8"</span>, <span class="st">"RPGRIP1L"</span>, <span class="st">"SAV1"</span>, <span class="st">"SDC1"</span>, <span class="st">"SDC3"</span>,</span>
<span>    <span class="st">"SEC16B"</span>, <span class="st">"SFPQ"</span>, <span class="st">"SFRP5"</span>, <span class="st">"SIPA1L1"</span>, <span class="st">"SLC2A14"</span>, <span class="st">"SLC6A9"</span>,</span>
<span>    <span class="st">"SPATA5L1"</span>, <span class="st">"SPINT1"</span>, <span class="st">"STAR"</span>, <span class="st">"STXBP5"</span>, <span class="st">"SUN3"</span>, <span class="st">"TACC2"</span>,</span>
<span>    <span class="st">"TACR1"</span>, <span class="st">"TAGLN2"</span>, <span class="st">"THPO"</span>, <span class="st">"TNIP1"</span>, <span class="st">"TP53"</span>, <span class="st">"TRMT2B"</span>, <span class="st">"TUBB1"</span>,</span>
<span>    <span class="st">"VDAC1"</span>, <span class="st">"VSIG8"</span>, <span class="st">"WNT3A"</span>, <span class="st">"WWOX"</span>, <span class="st">"XRCC4"</span>, <span class="st">"YME1L1"</span>,</span>
<span>    <span class="st">"ZBTB11"</span>, <span class="st">"ZSCAN21"</span></span>
<span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">50</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># make sure we have 100 genes</span></span>
<span><span class="va">smallSubset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">smallSubset</span>, <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">xdataRaw</span><span class="op">)</span>, <span class="fl">51</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">xdata</span> <span class="op">&lt;-</span> <span class="va">xdataRaw</span><span class="op">[</span>, <span class="va">smallSubset</span><span class="op">[</span><span class="va">smallSubset</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">xdataRaw</span><span class="op">)</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">ydata</span> <span class="op">&lt;-</span> <span class="va">ydataRaw</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">time</span>, <span class="va">status</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">time</span><span class="op">)</span> <span class="op">|</span> <span class="va">time</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span></span></code></pre></div>
<ul>
<li>Build dataset that overlaps with STRING data</li>
</ul>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#</span></span>
<span><span class="co"># Add degree 0 to genes not in STRING network</span></span>
<span></span>
<span><span class="va">myDegree</span> <span class="op">&lt;-</span> <span class="va">degreeNetworkVector</span><span class="op">[</span><span class="va">smallSubset</span><span class="op">]</span></span>
<span><span class="va">myString</span> <span class="op">&lt;-</span> <span class="va">stringNetworkBinary</span><span class="op">[</span><span class="va">smallSubset</span>, <span class="va">smallSubset</span><span class="op">]</span></span></code></pre></div>
<p>Degree distribution for sample set of gene features <em>(in
xdata)</em>.</p>
<p><img src="example_brca_protein-protein-interactions_survival_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<div class="section level3">
<h3 id="select-balanced-folds-for-cross-validation">Select balanced folds for cross-validation<a class="anchor" aria-label="anchor" href="#select-balanced-folds-for-cross-validation"></a>
</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="va">params</span><span class="op">$</span><span class="va">seed</span><span class="op">)</span></span>
<span><span class="va">foldid</span> <span class="op">&lt;-</span> <span class="fu">glmSparseNet</span><span class="fu">:::</span><span class="fu"><a href="../reference/balancedCvFolds.html">balancedCvFolds</a></span><span class="op">(</span><span class="va">ydata</span><span class="op">$</span><span class="va">status</span><span class="op">)</span><span class="op">$</span><span class="va">output</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="glmhub-model">glmHub model<a class="anchor" aria-label="anchor" href="#glmhub-model"></a>
</h3>
<p>Penalizes using the Hub heuristics, see <code>hubHeuristic</code>
function definition for more details.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">resultCVHub</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cv.glmSparseNet.html">cv.glmHub</a></span><span class="op">(</span><span class="va">xdata</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/survival/man/Surv.html" class="external-link">Surv</a></span><span class="op">(</span><span class="va">ydata</span><span class="op">$</span><span class="va">time</span>, <span class="va">ydata</span><span class="op">$</span><span class="va">status</span><span class="op">)</span>,</span>
<span>    family <span class="op">=</span> <span class="st">"cox"</span>,</span>
<span>    foldid <span class="op">=</span> <span class="va">foldid</span>,</span>
<span>    network <span class="op">=</span> <span class="va">myString</span>,</span>
<span>    network.options <span class="op">=</span> <span class="fu"><a href="../reference/networkOptions.html">networkOptions</a></span><span class="op">(</span>minDegree <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Kaplan-Meier estimator separating individuals by low and high risk
<em>(based on model’s coefficients)</em></p>
<pre><code><span><span class="co">## Warning: The `plot.title` argument of `separate2GroupsCox()` is deprecated as of</span></span>
<span><span class="co">## glmSparseNet 1.21.0.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Please use the `plotTitle` argument instead.</span></span>
<span><span class="co">## <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">## <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">## <span style="color: #555555;">generated.</span></span></span></code></pre>
<pre><code><span><span class="co">## Warning: The `legend.outside` argument of `separate2GroupsCox()` is deprecated as of</span></span>
<span><span class="co">## glmSparseNet 1.21.0.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Please use the `legendOutside` argument instead.</span></span>
<span><span class="co">## <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">## <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">## <span style="color: #555555;">generated.</span></span></span></code></pre>
<pre><code><span><span class="co">## $pvalue</span></span>
<span><span class="co">## [1] 9.628346e-05</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $plot</span></span></code></pre>
<p><img src="example_brca_protein-protein-interactions_survival_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## $km</span></span>
<span><span class="co">## Call: survfit(formula = survival::Surv(time, status) ~ group, data = prognosticIndexDf)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##                 n events median 0.95LCL 0.95UCL</span></span>
<span><span class="co">## Low risk - 1  983    129   3959    3738    7455</span></span>
<span><span class="co">## High risk - 1  97     23   2911    2469      NA</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="glmorphan-model">glmOrphan model<a class="anchor" aria-label="anchor" href="#glmorphan-model"></a>
</h3>
<p>Penalizes using the Orphan heuristics, see
<code>orphanHeuristic</code> function definition for more details.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">resultCVOrphan</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cv.glmSparseNet.html">cv.glmOrphan</a></span><span class="op">(</span><span class="va">xdata</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/survival/man/Surv.html" class="external-link">Surv</a></span><span class="op">(</span><span class="va">ydata</span><span class="op">$</span><span class="va">time</span>, <span class="va">ydata</span><span class="op">$</span><span class="va">status</span><span class="op">)</span>,</span>
<span>    family <span class="op">=</span> <span class="st">"cox"</span>,</span>
<span>    foldid <span class="op">=</span> <span class="va">foldid</span>,</span>
<span>    network <span class="op">=</span> <span class="va">myString</span>,</span>
<span>    network.options <span class="op">=</span> <span class="fu"><a href="../reference/networkOptions.html">networkOptions</a></span><span class="op">(</span>minDegree <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Kaplan-Meier estimator separating individuals by low and high risk
<em>(based on model’s coefficients)</em></p>
<pre><code><span><span class="co">## $pvalue</span></span>
<span><span class="co">## [1] 0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $plot</span></span></code></pre>
<p><img src="example_brca_protein-protein-interactions_survival_files/figure-html/unnamed-chunk-17-1.png" width="700"></p>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## $km</span></span>
<span><span class="co">## Call: survfit(formula = survival::Surv(time, status) ~ group, data = prognosticIndexDf)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##                 n events median 0.95LCL 0.95UCL</span></span>
<span><span class="co">## Low risk - 1  540     34   7455    6456      NA</span></span>
<span><span class="co">## High risk - 1 540    118   2866    2573    3472</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="elastic-net-model-without-network-penalization">Elastic Net model <em>(without network-penalization)</em><a class="anchor" aria-label="anchor" href="#elastic-net-model-without-network-penalization"></a>
</h3>
<p>Uses regular glmnet model as simple baseline</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://glmnet.stanford.edu" class="external-link">glmnet</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loaded glmnet 4.1-8</span></span></code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">resultCVGlmnet</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://glmnet.stanford.edu/reference/cv.glmnet.html" class="external-link">cv.glmnet</a></span><span class="op">(</span><span class="va">xdata</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/survival/man/Surv.html" class="external-link">Surv</a></span><span class="op">(</span><span class="va">ydata</span><span class="op">$</span><span class="va">time</span>, <span class="va">ydata</span><span class="op">$</span><span class="va">status</span><span class="op">)</span>,</span>
<span>    family <span class="op">=</span> <span class="st">"cox"</span>,</span>
<span>    foldid <span class="op">=</span> <span class="va">foldid</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Kaplan-Meier estimator separating individuals by low and high risk
<em>(based on model’s coefficients)</em></p>
<pre><code><span><span class="co">## $pvalue</span></span>
<span><span class="co">## [1] 0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $plot</span></span></code></pre>
<p><img src="example_brca_protein-protein-interactions_survival_files/figure-html/unnamed-chunk-19-1.png" width="700"></p>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## $km</span></span>
<span><span class="co">## Call: survfit(formula = survival::Surv(time, status) ~ group, data = prognosticIndexDf)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##                 n events median 0.95LCL 0.95UCL</span></span>
<span><span class="co">## Low risk - 1  540     38   6593    4398      NA</span></span>
<span><span class="co">## High risk - 1 540    114   2854    2551    3461</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="selected-genes">Selected genes<a class="anchor" aria-label="anchor" href="#selected-genes"></a>
</h3>
<p>Venn diagram of overlapping genes.</p>
<p><img src="example_brca_protein-protein-interactions_survival_files/figure-html/unnamed-chunk-20-1.png" width="700"></p>
<p>Descriptive table showing which genes are selected in each model</p>
<p>We can observe, that elastic net without network-based penalization
selects the best model with 40% more genes than glmOrphan and glmHub,
without loosing accuracy.</p>
<p><em>note</em>: size of circles represent the degree of that gene in
network.</p>
<p><img src="example_brca_protein-protein-interactions_survival_files/figure-html/unnamed-chunk-21-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.2 (2023-10-31)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">##  [1] stats4    grid      parallel  stats     graphics  grDevices utils    </span></span>
<span><span class="co">##  [8] datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] glmnet_4.1-8                glmSparseNet_1.21.2        </span></span>
<span><span class="co">##  [3] TCGAutils_1.22.2            curatedTCGAData_1.24.1     </span></span>
<span><span class="co">##  [5] MultiAssayExperiment_1.28.0 SummarizedExperiment_1.32.0</span></span>
<span><span class="co">##  [7] Biobase_2.62.0              GenomicRanges_1.54.1       </span></span>
<span><span class="co">##  [9] GenomeInfoDb_1.38.6         IRanges_2.36.0             </span></span>
<span><span class="co">## [11] S4Vectors_0.40.2            BiocGenerics_0.48.1        </span></span>
<span><span class="co">## [13] MatrixGenerics_1.14.0       matrixStats_1.2.0          </span></span>
<span><span class="co">## [15] VennDiagram_1.7.3           futile.logger_1.4.3        </span></span>
<span><span class="co">## [17] survival_3.5-7              reshape2_1.4.4             </span></span>
<span><span class="co">## [19] forcats_1.0.0               ggplot2_3.4.4              </span></span>
<span><span class="co">## [21] Matrix_1.6-1.1              dplyr_1.1.4                </span></span>
<span><span class="co">## [23] BiocStyle_2.30.0           </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] shape_1.4.6                   jsonlite_1.8.8               </span></span>
<span><span class="co">##   [3] magrittr_2.0.3                GenomicFeatures_1.54.3       </span></span>
<span><span class="co">##   [5] farver_2.1.1                  rmarkdown_2.25               </span></span>
<span><span class="co">##   [7] fs_1.6.3                      BiocIO_1.12.0                </span></span>
<span><span class="co">##   [9] zlibbioc_1.48.0               ragg_1.2.7                   </span></span>
<span><span class="co">##  [11] vctrs_0.6.5                   memoise_2.0.1                </span></span>
<span><span class="co">##  [13] Rsamtools_2.18.0              RCurl_1.98-1.14              </span></span>
<span><span class="co">##  [15] rstatix_0.7.2                 BiocBaseUtils_1.4.0          </span></span>
<span><span class="co">##  [17] htmltools_0.5.7               S4Arrays_1.2.0               </span></span>
<span><span class="co">##  [19] progress_1.2.3                AnnotationHub_3.10.0         </span></span>
<span><span class="co">##  [21] lambda.r_1.2.4                curl_5.2.0                   </span></span>
<span><span class="co">##  [23] broom_1.0.5                   SparseArray_1.2.4            </span></span>
<span><span class="co">##  [25] sass_0.4.8                    bslib_0.6.1                  </span></span>
<span><span class="co">##  [27] desc_1.4.3                    plyr_1.8.9                   </span></span>
<span><span class="co">##  [29] zoo_1.8-12                    futile.options_1.0.1         </span></span>
<span><span class="co">##  [31] cachem_1.0.8                  GenomicAlignments_1.38.2     </span></span>
<span><span class="co">##  [33] iterators_1.0.14              mime_0.12                    </span></span>
<span><span class="co">##  [35] lifecycle_1.0.4               pkgconfig_2.0.3              </span></span>
<span><span class="co">##  [37] R6_2.5.1                      fastmap_1.1.1                </span></span>
<span><span class="co">##  [39] GenomeInfoDbData_1.2.11       shiny_1.8.0                  </span></span>
<span><span class="co">##  [41] digest_0.6.34                 colorspace_2.1-0             </span></span>
<span><span class="co">##  [43] AnnotationDbi_1.64.1          ExperimentHub_2.10.0         </span></span>
<span><span class="co">##  [45] textshaping_0.3.7             RSQLite_2.3.5                </span></span>
<span><span class="co">##  [47] ggpubr_0.6.0                  labeling_0.4.3               </span></span>
<span><span class="co">##  [49] filelock_1.0.3                km.ci_0.5-6                  </span></span>
<span><span class="co">##  [51] fansi_1.0.6                   httr_1.4.7                   </span></span>
<span><span class="co">##  [53] abind_1.4-5                   compiler_4.3.2               </span></span>
<span><span class="co">##  [55] bit64_4.0.5                   withr_3.0.0                  </span></span>
<span><span class="co">##  [57] backports_1.4.1               BiocParallel_1.36.0          </span></span>
<span><span class="co">##  [59] carData_3.0-5                 DBI_1.2.1                    </span></span>
<span><span class="co">##  [61] highr_0.10                    ggsignif_0.6.4               </span></span>
<span><span class="co">##  [63] biomaRt_2.58.2                rappdirs_0.3.3               </span></span>
<span><span class="co">##  [65] DelayedArray_0.28.0           rjson_0.2.21                 </span></span>
<span><span class="co">##  [67] tools_4.3.2                   interactiveDisplayBase_1.40.0</span></span>
<span><span class="co">##  [69] httpuv_1.6.14                 glue_1.7.0                   </span></span>
<span><span class="co">##  [71] restfulr_0.0.15               promises_1.2.1               </span></span>
<span><span class="co">##  [73] checkmate_2.3.1               generics_0.1.3               </span></span>
<span><span class="co">##  [75] gtable_0.3.4                  KMsurv_0.1-5                 </span></span>
<span><span class="co">##  [77] tzdb_0.4.0                    tidyr_1.3.1                  </span></span>
<span><span class="co">##  [79] survminer_0.4.9               data.table_1.15.0            </span></span>
<span><span class="co">##  [81] hms_1.1.3                     car_3.1-2                    </span></span>
<span><span class="co">##  [83] xml2_1.3.6                    utf8_1.2.4                   </span></span>
<span><span class="co">##  [85] XVector_0.42.0                foreach_1.5.2                </span></span>
<span><span class="co">##  [87] BiocVersion_3.18.1            pillar_1.9.0                 </span></span>
<span><span class="co">##  [89] stringr_1.5.1                 later_1.3.2                  </span></span>
<span><span class="co">##  [91] splines_4.3.2                 BiocFileCache_2.10.1         </span></span>
<span><span class="co">##  [93] lattice_0.21-9                rtracklayer_1.62.0           </span></span>
<span><span class="co">##  [95] bit_4.0.5                     tidyselect_1.2.0             </span></span>
<span><span class="co">##  [97] Biostrings_2.70.2             knitr_1.45                   </span></span>
<span><span class="co">##  [99] gridExtra_2.3                 bookdown_0.37                </span></span>
<span><span class="co">## [101] xfun_0.42                     stringi_1.8.3                </span></span>
<span><span class="co">## [103] yaml_2.3.8                    evaluate_0.23                </span></span>
<span><span class="co">## [105] codetools_0.2-19              tibble_3.2.1                 </span></span>
<span><span class="co">## [107] BiocManager_1.30.22           cli_3.6.2                    </span></span>
<span><span class="co">## [109] xtable_1.8-4                  systemfonts_1.0.5            </span></span>
<span><span class="co">## [111] munsell_0.5.0                 jquerylib_0.1.4              </span></span>
<span><span class="co">## [113] survMisc_0.5.6                Rcpp_1.0.12                  </span></span>
<span><span class="co">## [115] GenomicDataCommons_1.26.0     dbplyr_2.4.0                 </span></span>
<span><span class="co">## [117] png_0.1-8                     XML_3.99-0.16.1              </span></span>
<span><span class="co">## [119] ellipsis_0.3.2                pkgdown_2.0.7                </span></span>
<span><span class="co">## [121] readr_2.1.5                   blob_1.2.4                   </span></span>
<span><span class="co">## [123] prettyunits_1.2.0             bitops_1.0-7                 </span></span>
<span><span class="co">## [125] scales_1.3.0                  purrr_1.0.2                  </span></span>
<span><span class="co">## [127] crayon_1.5.2                  rlang_1.1.3                  </span></span>
<span><span class="co">## [129] rvest_1.0.4                   KEGGREST_1.42.0              </span></span>
<span><span class="co">## [131] formatR_1.14</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by André Veríssimo, Susana Vinga.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
