<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Example for Classification Data -- Breast Invasive Carcinoma • glmSparseNet</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Example for Classification Data -- Breast Invasive Carcinoma">
<meta property="og:description" content="glmSparseNet">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">glmSparseNet</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.21.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/example_brca_logistic.html">Example for Classification Data -- Breast Invasive Carcinoma</a>
    </li>
    <li>
      <a href="../articles/example_brca_protein-protein-interactions_survival.html">Breast survival dataset using network from STRING DB</a>
    </li>
    <li>
      <a href="../articles/example_brca_survival.html">Example for Survival Data -- Breast Invasive Carcinoma</a>
    </li>
    <li>
      <a href="../articles/example_prad_survival.html">Example for Survival Data -- Prostate Adenocarcinoma</a>
    </li>
    <li>
      <a href="../articles/example_skcm_survival.html">Example for Survival Data -- Skin Melanoma</a>
    </li>
    <li>
      <a href="../articles/separate2GroupsCox.html">Separate 2 groups in Cox regression</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Example for Classification Data – Breast
Invasive Carcinoma</h1>
                        <h4 data-toc-skip class="author">Marta Lopes and
André Veríssimo</h4>
            
            <h4 data-toc-skip class="date">2024-02-14</h4>
      
      
      <div class="hidden name"><code>example_brca_logistic.Rmd</code></div>

    </div>

    
    
<div class="section level3">
<h3 id="instalation">Instalation<a class="anchor" aria-label="anchor" href="#instalation"></a>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"glmSparseNet"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="required-packages">Required Packages<a class="anchor" aria-label="anchor" href="#required-packages"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/therneau/survival" class="external-link">survival</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">futile.logger</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">curatedTCGAData</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://waldronlab.io/MultiAssayExperiment/" class="external-link">MultiAssayExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">TCGAutils</span><span class="op">)</span></span>
<span><span class="co">#</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://www.github.com/sysbiomed/glmSparseNet" class="external-link">glmSparseNet</a></span><span class="op">)</span></span>
<span><span class="co">#</span></span>
<span><span class="co"># Some general options for futile.logger the debugging package</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/futile.logger/man/flog.layout.html" class="external-link">flog.layout</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/futile.logger/man/flog.layout.html" class="external-link">layout.format</a></span><span class="op">(</span><span class="st">"[~l] ~m"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span></span>
<span>    <span class="st">"glmSparseNet.show_message"</span> <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    <span class="st">"glmSparseNet.base_dir"</span> <span class="op">=</span> <span class="fu">withr</span><span class="fu">::</span><span class="fu"><a href="https://withr.r-lib.org/reference/with_tempfile.html" class="external-link">local_tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co"># Setting ggplot2 default theme as minimal</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme_get.html" class="external-link">theme_set</a></span><span class="op">(</span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-data">Load data<a class="anchor" aria-label="anchor" href="#load-data"></a>
</h2>
<p>The data is loaded from an online curated dataset downloaded from
TCGA using <code>curatedTCGAData</code> bioconductor package and
processed.</p>
<p>To accelerate the process we use a very reduced dataset down to 107
variables only <em>(genes)</em>, which is stored as a data object in
this package. However, the procedure to obtain the data manually is
described in the following chunk.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">brca</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/curatedTCGAData/man/curatedTCGAData.html" class="external-link">curatedTCGAData</a></span><span class="op">(</span></span>
<span>    diseaseCode <span class="op">=</span> <span class="st">"BRCA"</span>, assays <span class="op">=</span> <span class="st">"RNASeq2GeneNorm"</span>,</span>
<span>    version <span class="op">=</span> <span class="st">"1.1.38"</span>, dry.run <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">brca</span> <span class="op">&lt;-</span> <span class="fu">TCGAutils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/TCGAutils/man/curatedTCGAData-helpers.html" class="external-link">TCGAsplitAssays</a></span><span class="op">(</span><span class="va">brca</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"01"</span>, <span class="st">"11"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">xdataRaw</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">brca</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">brca</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get matches between survival and assay data</span></span>
<span><span class="va">classV</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/TCGAutils/man/TCGAbiospec.html" class="external-link">TCGAbiospec</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">xdataRaw</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">sample_definition</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">classV</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">xdataRaw</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># keep features with standard deviation &gt; 0</span></span>
<span><span class="va">xdataRaw</span> <span class="op">&lt;-</span> <span class="va">xdataRaw</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">xdataRaw</span>, <span class="fl">2</span>, <span class="va">sd</span><span class="op">)</span> <span class="op">!=</span> <span class="fl">0</span><span class="op">]</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/scale.html" class="external-link">scale</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="va">params</span><span class="op">$</span><span class="va">seed</span><span class="op">)</span></span>
<span><span class="va">smallSubset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="st">"CD5"</span>, <span class="st">"CSF2RB"</span>, <span class="st">"HSF1"</span>, <span class="st">"IRGC"</span>, <span class="st">"LRRC37A6P"</span>, <span class="st">"NEUROG2"</span>,</span>
<span>    <span class="st">"NLRC4"</span>, <span class="st">"PDE11A"</span>, <span class="st">"PIK3CB"</span>, <span class="st">"QARS"</span>, <span class="st">"RPGRIP1L"</span>, <span class="st">"SDC1"</span>,</span>
<span>    <span class="st">"TMEM31"</span>, <span class="st">"YME1L1"</span>, <span class="st">"ZBTB11"</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">xdataRaw</span><span class="op">)</span>, <span class="fl">100</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">xdata</span> <span class="op">&lt;-</span> <span class="va">xdataRaw</span><span class="op">[</span>, <span class="va">smallSubset</span><span class="op">[</span><span class="va">smallSubset</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">xdataRaw</span><span class="op">)</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">ydata</span> <span class="op">&lt;-</span> <span class="va">classV</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="fit-models">Fit models<a class="anchor" aria-label="anchor" href="#fit-models"></a>
</h2>
<p>Fit model model penalizing by the hubs using the cross-validation
function by <code>cv.glmHub</code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fitted</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cv.glmSparseNet.html">cv.glmHub</a></span><span class="op">(</span><span class="va">xdata</span>, <span class="va">ydata</span>,</span>
<span>    family <span class="op">=</span> <span class="st">"binomial"</span>,</span>
<span>    network <span class="op">=</span> <span class="st">"correlation"</span>,</span>
<span>    nlambda <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>    options <span class="op">=</span> <span class="fu"><a href="../reference/networkOptions.html">networkOptions</a></span><span class="op">(</span></span>
<span>        cutoff <span class="op">=</span> <span class="fl">.6</span>,</span>
<span>        minDegree <span class="op">=</span> <span class="fl">.2</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="results-of-cross-validation">Results of Cross Validation<a class="anchor" aria-label="anchor" href="#results-of-cross-validation"></a>
</h2>
<p>Shows the results of <code>1000</code> different parameters used to
find the optimal value in 10-fold cross-validation. The two vertical
dotted lines represent the best model and a model with less variables
selected <em>(genes)</em>, but within a standard error distance from the
best.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">fitted</span><span class="op">)</span></span></code></pre></div>
<p><img src="example_brca_logistic_files/figure-html/results-1.png" width="700"></p>
<div class="section level3">
<h3 id="coefficients-of-selected-model-from-cross-validation">Coefficients of selected model from Cross-Validation<a class="anchor" aria-label="anchor" href="#coefficients-of-selected-model-from-cross-validation"></a>
</h3>
<p>Taking the best model described by <code>lambda.min</code></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">coefsCV</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/funprog.html" class="external-link">Filter</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="va">.x</span> <span class="op">!=</span> <span class="fl">0</span>, <span class="fu"><a href="https://rdrr.io/r/stats/coef.html" class="external-link">coef</a></span><span class="op">(</span><span class="va">fitted</span>, s <span class="op">=</span> <span class="st">"lambda.min"</span><span class="op">)</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>    ensembl.id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">coefsCV</span><span class="op">)</span>,</span>
<span>    gene.name <span class="op">=</span> <span class="fu"><a href="../reference/geneNames.html">geneNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">coefsCV</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">external_gene_name</span>,</span>
<span>    coefficient <span class="op">=</span> <span class="va">coefsCV</span>,</span>
<span>    stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">gene.name</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="left">ensembl.id</th>
<th align="left">gene.name</th>
<th align="right">coefficient</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">(Intercept)</td>
<td align="left">(Intercept)</td>
<td align="left">(Intercept)</td>
<td align="right">-6.8189813</td>
</tr>
<tr class="even">
<td align="left">AMOTL1</td>
<td align="left">AMOTL1</td>
<td align="left">AMOTL1</td>
<td align="right">0.4430643</td>
</tr>
<tr class="odd">
<td align="left">ATR</td>
<td align="left">ATR</td>
<td align="left">ATR</td>
<td align="right">1.2498304</td>
</tr>
<tr class="even">
<td align="left">B3GALT2</td>
<td align="left">B3GALT2</td>
<td align="left">B3GALT2</td>
<td align="right">-0.0867011</td>
</tr>
<tr class="odd">
<td align="left">BAG2</td>
<td align="left">BAG2</td>
<td align="left">BAG2</td>
<td align="right">-0.1841676</td>
</tr>
<tr class="even">
<td align="left">C16orf82</td>
<td align="left">C16orf82</td>
<td align="left">C16orf82</td>
<td align="right">0.0396368</td>
</tr>
<tr class="odd">
<td align="left">CD5</td>
<td align="left">CD5</td>
<td align="left">CD5</td>
<td align="right">-1.1200445</td>
</tr>
<tr class="even">
<td align="left">CIITA</td>
<td align="left">CIITA</td>
<td align="left">CIITA</td>
<td align="right">0.4256103</td>
</tr>
<tr class="odd">
<td align="left">DCP1A</td>
<td align="left">DCP1A</td>
<td align="left">DCP1A</td>
<td align="right">0.2994599</td>
</tr>
<tr class="even">
<td align="left">FAM86B1</td>
<td align="left">FAM86B1</td>
<td align="left">FAM86B1</td>
<td align="right">0.2025463</td>
</tr>
<tr class="odd">
<td align="left">FNIP2</td>
<td align="left">FNIP2</td>
<td align="left">FNIP2</td>
<td align="right">0.6101759</td>
</tr>
<tr class="even">
<td align="left">GDF11</td>
<td align="left">GDF11</td>
<td align="left">GDF11</td>
<td align="right">-0.2676642</td>
</tr>
<tr class="odd">
<td align="left">GNG11</td>
<td align="left">GNG11</td>
<td align="left">GNG11</td>
<td align="right">3.0659066</td>
</tr>
<tr class="even">
<td align="left">GREM2</td>
<td align="left">GREM2</td>
<td align="left">GREM2</td>
<td align="right">-0.2014884</td>
</tr>
<tr class="odd">
<td align="left">GZMB</td>
<td align="left">GZMB</td>
<td align="left">GZMB</td>
<td align="right">-2.7663574</td>
</tr>
<tr class="even">
<td align="left">HAX1</td>
<td align="left">HAX1</td>
<td align="left">HAX1</td>
<td align="right">-0.1516837</td>
</tr>
<tr class="odd">
<td align="left">IL2</td>
<td align="left">IL2</td>
<td align="left">IL2</td>
<td align="right">0.6327083</td>
</tr>
<tr class="even">
<td align="left">MMP28</td>
<td align="left">MMP28</td>
<td align="left">MMP28</td>
<td align="right">-0.8438024</td>
</tr>
<tr class="odd">
<td align="left">MS4A4A</td>
<td align="left">MS4A4A</td>
<td align="left">MS4A4A</td>
<td align="right">1.1614779</td>
</tr>
<tr class="even">
<td align="left">NDRG2</td>
<td align="left">NDRG2</td>
<td align="left">NDRG2</td>
<td align="right">1.1142519</td>
</tr>
<tr class="odd">
<td align="left">NLRC4</td>
<td align="left">NLRC4</td>
<td align="left">NLRC4</td>
<td align="right">-1.4434578</td>
</tr>
<tr class="even">
<td align="left">PIK3CB</td>
<td align="left">PIK3CB</td>
<td align="left">PIK3CB</td>
<td align="right">-0.3880002</td>
</tr>
<tr class="odd">
<td align="left">ZBTB11</td>
<td align="left">ZBTB11</td>
<td align="left">ZBTB11</td>
<td align="right">-0.3325729</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="accuracy">Accuracy<a class="anchor" aria-label="anchor" href="#accuracy"></a>
</h3>
<pre><code><span><span class="co">## [INFO] Misclassified (11)</span></span></code></pre>
<pre><code><span><span class="co">## [INFO]   * False primary solid tumour: 7</span></span></code></pre>
<pre><code><span><span class="co">## [INFO]   * False normal              : 4</span></span></code></pre>
<p>Histogram of predicted response</p>
<p><img src="example_brca_logistic_files/figure-html/predict-1.png" width="700"></p>
<p>ROC curve</p>
<pre><code><span><span class="co">## Setting levels: control = Primary Solid Tumor, case = Solid Tissue Normal</span></span></code></pre>
<pre><code><span><span class="co">## Setting direction: controls &lt; cases</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Please use `linewidth` instead.</span></span>
<span><span class="co">## <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">## <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">## <span style="color: #555555;">generated.</span></span></span></code></pre>
<p><img src="example_brca_logistic_files/figure-html/roc-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.2 (2023-10-31)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] glmSparseNet_1.21.2         TCGAutils_1.22.2           </span></span>
<span><span class="co">##  [3] curatedTCGAData_1.24.1      MultiAssayExperiment_1.28.0</span></span>
<span><span class="co">##  [5] SummarizedExperiment_1.32.0 Biobase_2.62.0             </span></span>
<span><span class="co">##  [7] GenomicRanges_1.54.1        GenomeInfoDb_1.38.6        </span></span>
<span><span class="co">##  [9] IRanges_2.36.0              S4Vectors_0.40.2           </span></span>
<span><span class="co">## [11] BiocGenerics_0.48.1         MatrixGenerics_1.14.0      </span></span>
<span><span class="co">## [13] matrixStats_1.2.0           futile.logger_1.4.3        </span></span>
<span><span class="co">## [15] survival_3.5-7              ggplot2_3.4.4              </span></span>
<span><span class="co">## [17] dplyr_1.1.4                 BiocStyle_2.30.0           </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] shape_1.4.6                   jsonlite_1.8.8               </span></span>
<span><span class="co">##   [3] magrittr_2.0.3                GenomicFeatures_1.54.3       </span></span>
<span><span class="co">##   [5] farver_2.1.1                  rmarkdown_2.25               </span></span>
<span><span class="co">##   [7] fs_1.6.3                      BiocIO_1.12.0                </span></span>
<span><span class="co">##   [9] zlibbioc_1.48.0               ragg_1.2.7                   </span></span>
<span><span class="co">##  [11] vctrs_0.6.5                   memoise_2.0.1                </span></span>
<span><span class="co">##  [13] Rsamtools_2.18.0              RCurl_1.98-1.14              </span></span>
<span><span class="co">##  [15] htmltools_0.5.7               S4Arrays_1.2.0               </span></span>
<span><span class="co">##  [17] BiocBaseUtils_1.4.0           progress_1.2.3               </span></span>
<span><span class="co">##  [19] AnnotationHub_3.10.0          lambda.r_1.2.4               </span></span>
<span><span class="co">##  [21] curl_5.2.0                    pROC_1.18.5                  </span></span>
<span><span class="co">##  [23] SparseArray_1.2.4             sass_0.4.8                   </span></span>
<span><span class="co">##  [25] bslib_0.6.1                   desc_1.4.3                   </span></span>
<span><span class="co">##  [27] plyr_1.8.9                    futile.options_1.0.1         </span></span>
<span><span class="co">##  [29] cachem_1.0.8                  GenomicAlignments_1.38.2     </span></span>
<span><span class="co">##  [31] iterators_1.0.14              mime_0.12                    </span></span>
<span><span class="co">##  [33] lifecycle_1.0.4               pkgconfig_2.0.3              </span></span>
<span><span class="co">##  [35] Matrix_1.6-1.1                R6_2.5.1                     </span></span>
<span><span class="co">##  [37] fastmap_1.1.1                 GenomeInfoDbData_1.2.11      </span></span>
<span><span class="co">##  [39] shiny_1.8.0                   digest_0.6.34                </span></span>
<span><span class="co">##  [41] colorspace_2.1-0              AnnotationDbi_1.64.1         </span></span>
<span><span class="co">##  [43] ExperimentHub_2.10.0          textshaping_0.3.7            </span></span>
<span><span class="co">##  [45] RSQLite_2.3.5                 labeling_0.4.3               </span></span>
<span><span class="co">##  [47] filelock_1.0.3                fansi_1.0.6                  </span></span>
<span><span class="co">##  [49] httr_1.4.7                    abind_1.4-5                  </span></span>
<span><span class="co">##  [51] compiler_4.3.2                bit64_4.0.5                  </span></span>
<span><span class="co">##  [53] withr_3.0.0                   backports_1.4.1              </span></span>
<span><span class="co">##  [55] BiocParallel_1.36.0           DBI_1.2.1                    </span></span>
<span><span class="co">##  [57] highr_0.10                    biomaRt_2.58.2               </span></span>
<span><span class="co">##  [59] rappdirs_0.3.3                DelayedArray_0.28.0          </span></span>
<span><span class="co">##  [61] rjson_0.2.21                  tools_4.3.2                  </span></span>
<span><span class="co">##  [63] interactiveDisplayBase_1.40.0 httpuv_1.6.14                </span></span>
<span><span class="co">##  [65] glue_1.7.0                    restfulr_0.0.15              </span></span>
<span><span class="co">##  [67] promises_1.2.1                grid_4.3.2                   </span></span>
<span><span class="co">##  [69] checkmate_2.3.1               generics_0.1.3               </span></span>
<span><span class="co">##  [71] gtable_0.3.4                  tzdb_0.4.0                   </span></span>
<span><span class="co">##  [73] hms_1.1.3                     xml2_1.3.6                   </span></span>
<span><span class="co">##  [75] utf8_1.2.4                    XVector_0.42.0               </span></span>
<span><span class="co">##  [77] foreach_1.5.2                 BiocVersion_3.18.1           </span></span>
<span><span class="co">##  [79] pillar_1.9.0                  stringr_1.5.1                </span></span>
<span><span class="co">##  [81] later_1.3.2                   splines_4.3.2                </span></span>
<span><span class="co">##  [83] BiocFileCache_2.10.1          lattice_0.21-9               </span></span>
<span><span class="co">##  [85] rtracklayer_1.62.0            bit_4.0.5                    </span></span>
<span><span class="co">##  [87] tidyselect_1.2.0              Biostrings_2.70.2            </span></span>
<span><span class="co">##  [89] knitr_1.45                    bookdown_0.37                </span></span>
<span><span class="co">##  [91] xfun_0.42                     stringi_1.8.3                </span></span>
<span><span class="co">##  [93] yaml_2.3.8                    evaluate_0.23                </span></span>
<span><span class="co">##  [95] codetools_0.2-19              tibble_3.2.1                 </span></span>
<span><span class="co">##  [97] BiocManager_1.30.22           cli_3.6.2                    </span></span>
<span><span class="co">##  [99] xtable_1.8-4                  systemfonts_1.0.5            </span></span>
<span><span class="co">## [101] munsell_0.5.0                 jquerylib_0.1.4              </span></span>
<span><span class="co">## [103] Rcpp_1.0.12                   GenomicDataCommons_1.26.0    </span></span>
<span><span class="co">## [105] dbplyr_2.4.0                  png_0.1-8                    </span></span>
<span><span class="co">## [107] XML_3.99-0.16.1               parallel_4.3.2               </span></span>
<span><span class="co">## [109] ellipsis_0.3.2                pkgdown_2.0.7                </span></span>
<span><span class="co">## [111] readr_2.1.5                   blob_1.2.4                   </span></span>
<span><span class="co">## [113] prettyunits_1.2.0             bitops_1.0-7                 </span></span>
<span><span class="co">## [115] glmnet_4.1-8                  scales_1.3.0                 </span></span>
<span><span class="co">## [117] purrr_1.0.2                   crayon_1.5.2                 </span></span>
<span><span class="co">## [119] rlang_1.1.3                   KEGGREST_1.42.0              </span></span>
<span><span class="co">## [121] rvest_1.0.4                   formatR_1.14</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by André Veríssimo, Susana Vinga.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
